src/keys.cc

```diff
@@ -102,9 +103,41 @@ KeyList parse_keys(StringView str)
         auto name_it = find_if(keynamemap, [&desc](const KeyAndName& item)
                                            { return item.name == desc; });
         if (name_it != std::end(keynamemap))
-            result.push_back(canonicalize_ifn({ modifier, name_it->key }));
+        {
+            Codepoint key = name_it->key;
+
+            if (modifier & Key::Modifiers::Shift)
+            {
+                if (key < 0xD800 || key > 0xDFFF)
+                {
+                    if (!is_lower(name_it->key))
```

Comment:
we can probably tolerate upper case as well `<s-X> == <s-x> == <X>`, use `is_basic_alpha` to check that.